app/products/playbooks/screens/playbook_run/rename_playbook_run_bottom_sheet.tsx

```diff
@@ -0,0 +1,105 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback, useEffect, useState} from 'react';
+import {useIntl} from 'react-intl';
+import {Keyboard, StyleSheet, View} from 'react-native';
+
+import FloatingTextInput from '@components/floating_input/floating_text_input_label';
+import {useTheme} from '@context/theme';
+import useAndroidHardwareBackHandler from '@hooks/android_back_handler';
+import useNavButtonPressed from '@hooks/navigation_button_pressed';
+import SecurityManager from '@managers/security_manager';
+import {buildNavigationButton, popTopScreen, setButtons} from '@screens/navigation';
+
+import type {AvailableScreens} from '@typings/screens/navigation';
+
+type Props = {
+    componentId: AvailableScreens;
+    currentTitle: string;
+    onSave: (newTitle: string) => void;
+}
+
+const SAVE_BUTTON_ID = 'save-playbook-run-name';
+
+const close = (componentId: AvailableScreens): void => {
+    Keyboard.dismiss();
+    popTopScreen(componentId);
+};
+
+const styles = StyleSheet.create({
+    container: {
+        flex: 1,
+        paddingVertical: 32,
+        paddingHorizontal: 20,
+    },
+});
+
+const RenamePlaybookRunBottomSheet = ({
+    componentId,
+    currentTitle,
+    onSave,
+}: Props) => {
+    const intl = useIntl();
+    const {formatMessage} = intl;
+    const theme = useTheme();
+
+    const [title, setTitle] = useState<string>(currentTitle);
+    const [canSave, setCanSave] = useState(false);
+
+    const rightButton = React.useMemo(() => {
+        const base = buildNavigationButton(
+            SAVE_BUTTON_ID,
+            'playbooks.playbook_run.rename.button',
+            undefined,
+            formatMessage({id: 'playbooks.playbook_run.rename.button', defaultMessage: 'Save'}),
+        );
+        base.enabled = canSave;
+        base.color = theme.sidebarHeaderTextColor;
+        return base;
+    }, [formatMessage, canSave, theme.sidebarHeaderTextColor]);
+
+    useEffect(() => {
+        setButtons(componentId, {
+            rightButtons: [rightButton],
+        });
+    }, [rightButton, componentId]);
+
+    useEffect(() => {
+        setCanSave(title.trim().length > 0 && title !== currentTitle);
+    }, [title, currentTitle]);
+
+    const handleClose = useCallback(() => {
+        close(componentId);
+    }, [componentId]);
+
+    const handleSave = useCallback(() => {
+        if (title.trim().length > 0) {
```

Comment:
I guess this should be `canSave`, which is checking more things than what this is checking.